tools/api_check/lib/apicheck.dart

```diff
@@ -35,7 +35,7 @@ List<String> getDartClassFields({
   final RegExp fieldExp = RegExp(r'_k(\w*)Index');
   final List<String> fields = <String>[];
   for (final CompilationUnitMember unitMember in result.unit.declarations) {
-    if (unitMember is ClassDeclaration && unitMember.name.lexeme == className) {
+    if ((unitMember is ClassDeclaration || unitMember is EnumDeclaration) && (unitMember as NamedCompilationUnitMember).name.lexeme == className) {
       for (final ClassMember classMember in unitMember.members) {
```

Comment:
Looks like deduplicating this is not as easy as I thought as there is no common base class of ClassDeclaration and EnumDeclaration that defines the `.members` here. We may have to come up with a different approach to de-duplicating this.

Reply:
If it doesn't work now... lol